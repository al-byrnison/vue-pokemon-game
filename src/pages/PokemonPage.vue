<template>
	<h1 v-if="!pokemon">Esperando...</h1>
	<div
		v-else
		class="container"
	>
		<h1>¿Quién es este Pokemon?</h1>
		<PokemonPicture
			:pokemonId="pokemon.id"
			:showPokemon="showPokemon"
		/>
		<PokemonOptions
			:pokemons="pokemonArr"
			@selectionPokemon="checkAnswer"
		/>
		<div
			v-if="showAnswer"
			class="answer-container"
		>
			<h2 class="fade-in centered">{{ message }}</h2>
			<img
				src="../assets/reload.svg"
				alt="pokemon"
				@click="newGame"
				class="btn-reload"
			/>
		</div>
	</div>
</template>

<script>
	import PokemonPicture from '@/components/PokemonPicture'
	import PokemonOptions from '@/components/PokemonOptions'
	import getPokemonOptions from '@/helpers/getPokemonOptions'

	export default {
		components: { PokemonPicture, PokemonOptions },
		data() {
			return {
				pokemonArr: [],
				pokemon: null,
				showPokemon: false,
				showAnswer: false,
				message: '',
			}
		},
		methods: {
			async mixPokemonArray() {
				this.pokemonArr = await getPokemonOptions()
				const randomInt = Math.floor(Math.random() * 4)
				this.pokemon = this.pokemonArr[randomInt]
			},
			checkAnswer(pokemonId) {
				this.showPokemon = true
				this.showAnswer = true
				if (pokemonId === this.pokemon.id) {
					this.message = `Correcto el pokemon es ${this.pokemon.name}`
				} else {
					this.message = `Oops, era ${this.pokemon.name}`
				}
			},
			newGame() {
				this.showPokemon = false
				this.showAnswer = false
				this.pokemonArr = []
				this.pokemon = null
				this.mixPokemonArray()
			},
		},
		mounted() {
			this.mixPokemonArray()
		},
	}
</script>

<style scoped>
	button {
		margin: 0 auto;
	}
	.answer-container {
		justify-content: center;
	}
	.btn-reload {
		width: 3rem;
		height: 3rem;
	}
</style>
