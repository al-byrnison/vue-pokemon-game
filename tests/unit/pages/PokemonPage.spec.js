import { mount, shallowMount } from '@vue/test-utils';
import PokemonPage from '@/pages/PokemonPage'
import { pokemons } from '../mocks/pokemons.mock';

describe('PokemonPage pages', () => {
    let wrapper
    beforeEach(() => {
        wrapper = shallowMount(PokemonPage)
    })

    test('should call mixpokemonArray onMount', () => {
      const mixPokemonArraySpy = jest.spyOn(PokemonPage.methods, 'mixPokemonArray')
        const wrapper = shallowMount(PokemonPage)
      expect(mixPokemonArraySpy).toHaveBeenCalled()
    })

    test('should snapshot match after mount component', () => {
      const wrapper = mount(PokemonPage, {
        data() {
            return {
                pokemonArr: pokemons,
				pokemon: pokemons[0],
				showPokemon: false,
				showAnswer: false,
				message: '',
            }
        }
      })
      expect(wrapper.html()).toMatchSnapshot()
    })
    
    
});
