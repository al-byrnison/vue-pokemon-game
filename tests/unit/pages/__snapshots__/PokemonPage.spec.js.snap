// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonPage pages should snapshot match after mount component 1`] = `
<div class="container">
  <h1>¿Quién es este Pokemon?</h1>
  <div class="pokemon-container"><img src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/1.svg" alt="pokemon" class="pokemon-hidden"></div>
  <div class="options-container">
    <ul>
      <li>bulbasaur</li>
      <li>ivysaur</li>
      <li>venusaur</li>
      <li>charmander</li>
    </ul>
  </div>
  <!--v-if-->
</div>
`;
